### ISSUE TITLE
Consider destructuring in 'Variables should not be self-assigned'

### ISSUE THREAD

**Issue by inverno** [SonarSource/SonarJS#802]
RSPEC-1656

We should catch this case :
```
let x = [];
x = [...x];
```

**Comment by vilchik-elena** [comment]
current rule implementation is provided by ESLint for ROI is low as we can't simply do the fix. Closing the issue
